src/otaclient_common/_env.py

```diff
@@ -45,3 +51,15 @@ def get_dynamic_client_chroot_path() -> Optional[str]:
     if is_dynamic_client_running():
         return cfg.DYNAMIC_CLIENT_MNT_HOST_ROOT
     return None
+
+
+@cache
+def get_otaclient_squashfs_download_dst() -> str:
+    """Get the location to hold downloaded otaclient squashfs image."""
+    if is_dynamic_client_running():
+        return replace_root(
+            cfg.DYNAMIC_CLIENT_SQUASHFS_FILE,
+            cfg.CANONICAL_ROOT,
+            cfg.DYNAMIC_CLIENT_MNT_HOST_ROOT,
+        )
+    return cfg.DYNAMIC_CLIENT_SQUASHFS_FILE
```

Comment:
When the otaclient originally runs as systemd managed app image, the downloaded otaclient squashfs image should be placed under `/host_root` instead of `/`.